time/src/main/java/net/jqwik/time/internal/properties/arbitraries/DefaultLocalDateArbitrary.java

```diff
@@ -70,21 +70,74 @@
 	}
 
 	private void checkIfValuesArePossible(LocalDate min, LocalDate max) {
-		//TODO: Methode muss getestet werden!!!!! - Besonderer Testfall: z.B. Februar erlaubt, aber nur 31.
 		if (max.getYear() - 2 >= min.getYear()) {
-			return;
-		}
-		for (LocalDate date = min; !date.isAfter(max); date = date.plusDays(1)) {
-			if (dayOfMonthBetween.getMin() >= date.getDayOfMonth()
-					&& dayOfMonthBetween.getMax() <= date.getDayOfMonth()
-					&& allowedMonths.get().contains(date.getMonth())
-			) {
+			if (valuesArePossible()) {
 				return;
 			}
+		} else {
+			int minDayOfMonth = dayOfMonthBetween.getMin() == null ? 1 : dayOfMonthBetween.getMin();
+			int maxDayOfMonth = dayOfMonthBetween.getMax() == null ? 31 : dayOfMonthBetween.getMax();
+			for (LocalDate date = min; !date.isAfter(max); date = date.plusDays(1)) {
+				if (minDayOfMonth <= date.getDayOfMonth()
+						&& maxDayOfMonth >= date.getDayOfMonth()
+						&& allowedMonths.get().contains(date.getMonth())
+				) {
+					return;
+				}
+			}
 		}
 		throw new IllegalArgumentException("These min/max configurations cannot be used together: No values are possible.");
 	}
 
+	private boolean valuesArePossible() {
```

Comment:
This looks terribly complicated. What happens if you just don't check? Won't the generator just fail at first attempt and tell that no value can be generated?